.circleci/config.yml

```diff
@@ -23,18 +28,35 @@ jobs:
         # Installing required packages for `make -C doc check command` to work.
       - run: sudo -E apt-get -yq update
       - run: sudo -E apt-get -yq --no-install-suggests --no-install-recommends --force-yes install dvipng texlive-latex-base texlive-latex-extra
+      - run:
+          name: Generating today's date and writing to file to generate daily new cache key
+          command: |
+            date +%F > today
       - restore_cache:
-          key: v1-packages+datasets-{{ .Branch }}
-      - run: wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh
-      - run: chmod +x ~/miniconda.sh && ~/miniconda.sh -b
+          key: v5-packages+datasets-{{ .Branch }}-{{ checksum "today" }}
+      - run:
+          name: Download & install conda if absent
+          command: |
+            if
+              ls $HOME/miniconda3/bin | grep conda -q
+            then
+              echo "(Mini)Conda already present from the cache."
+            else
+              wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh
+              chmod +x ~/miniconda.sh && ~/miniconda.sh -b
+            fi
       - run:
           name: Setup conda path in env variables
           command: |
             echo 'export PATH="$HOME/miniconda3/bin:$PATH"'  >> $BASH_ENV
       - run:
-          name: Create conda env
+          name: Create new conda env  # Separate command so if it fails (when using cache) the entire build will not.
           command: |
-            conda create -n testenv python=3.6 numpy scipy scikit-learn matplotlib pandas \
+            if conda env list | grep testenv ; then echo "Conda env testenv already exists courtesy of the cache."; else conda create -n testenv -yq; fi
```

Comment:
Long line (hard to review as it extends longer than the width of the page). Can you break it down?